integration-cli/docker_utils.go

```diff
@@ -569,17 +569,42 @@ func fakeContext(dockerfile string, files map[string]string) (*FakeContext, erro
 	return ctx, nil
 }
 
-type FakeStorage struct {
+// FakeStorage is a static file server. It might be running locally or remotely
+// on test host.
+type FakeStorage interface {
+	Close() error
+	URL() string
+	CtxDir() string
+}
+
+// fakeStorage returns either a local or remote (at daemon machine) file server
+func fakeStorage(files map[string]string) (FakeStorage, error) {
+	if isLocalDaemon {
```

Comment:
@tianon as a matter of fact, we can. but I think we shouldn't be building images and starting containers if not necessary. That would slow the tests down and we can do the same thing with local httptest server very well.

One can also argue that this tests the case remote URL is not a container, but I'm not sure how good this argument is. Anyway, there's no harm in maintaining both as you said.